src/bill.ts

```diff
@@ -182,11 +182,37 @@ export class BbitQRBillGenerator {
     newY = this._renderPayableTo(doc, options.paymentPartRightX, newY, options.paymentTitleFontSize, options.paymentFontSize, params);
     newY = this._renderReference(doc, options.paymentPartRightX, newY, options.paymentTitleFontSize, options.paymentFontSize, params);
 
-    if (params.unstructuredMessage && params.billInformation) {
+    if (params.unstructuredMessage || params.billInformation) {
       newY = newY + (options.paymentFontSize + 1) * 2;
       doc.fontSize(options.paymentTitleFontSize).font('Helvetica-Bold').text(this._t.additionalInfo, options.paymentPartRightX, newY);
-      const message = params.billInformation || params.unstructuredMessage;
-      for (const line of message.split('\n')) {
+
+      let message = '';
+
+      if (params.unstructuredMessage) {
```

Comment:
Looking at the docs it says:

```
If there are not enough characters available when a structured and unstructured message is delivered at the same time, the structured reference is to be preferred so that the automatic reconciliation of the accounting by the biller on the basis of the reference number (=structured message) remains guaranteed. In this case, the unstructured message should be shortened.
```

So I think the `billInformation` part should be rendered first and after this the `unstructuredMessage`, or am I wrong?

Reply:
Well, I've based my code on the "Table 3" of page 19 from : https://www.paymentstandards.ch/dam/downloads/ig-qr-bill-en.pdf

There it's written:

>  Both fields together can only contain a maximum of 140characters. If both elements are filled in, then a line break can be introduced after the information in the first element “Ustrd”(Unstructured message).

This sentence lets imagine the first element should be the unstructured message, so I've put in first position.

> If there is insufficient space, the line break can be omitted (but this makes it more difficult to read). If not all the details contained in the QR code can be displayed, the shortened content must be marked with an ellipsis “...”at the end. It must be ensured that all personal data is displayed.

There, they say we should just truncate the message and add an ellipsis.

---

Your quotation is about mapping table between QR Code and Swift message (Annex F), that's right ?

Is it meaning the implementation guideline says the human readable texts could be different from the qr encoded structure ? Do you understand it the same way ?

Reply:
You are absolutely right, I am just to dumb to read the document properly 🙈